core/vm/common.go

```diff
@@ -28,6 +28,8 @@ import (
 // Global Debug flag indicating Debug VM (full logging)
 var Debug bool
 
+var GenerateStructLogs bool = false
```

Comment:
@obscuren Couldn't we do this cleaner without adding a new global variable? In the long term we should really try to contain the modules to enable using Geth as a library, but these globals really make that hard. With the debug flag it's somewhat arguable that you don't mess with it, but if we expose this structured logger over an API call, it could produce very weird behaviors (e.g. concurrent API calls could arbitrarily turn this on and off; multiple VM executions could produce weird output for the user, etc). Of course these are still contained somewhat requiring a user to manually call the debug API, just saying it would be nicer if we could contain this without needing a global flag.